cmd/controller-manager/main.go

```diff
@@ -115,6 +115,11 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Duplicate detection across WDSes
```

Comment:
This is doing the check once, at controller startup. Is that really what you intended?